src/dsp_tools/commands/xmllib/different_approaches/regular_class.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import warnings
+from typing import Any
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.value_checkers import is_integer
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+
+"""
+Advantage: 
+    Validation always happens at creation.
+    Resource ID is communicated during a warning
+Disadvantage: 
+    __init__ is wordy
+"""
+
+
+class IntValue:
+    value: int | str
+    prop_name: str
+    permissions: str | None = "prop-default"
+    comment: str | None = None
+
+    def __init__(
```

Comment:
I would not use normal classes. If I'm not mistaken, the dataclass-approach with factory + `init=False` covers all advantages, without the disadvantage of the verbose __init__

Reply:
I honestly don't see the difference between `init=False` and using a normal class. Because you have to write the the values yourself too.